<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>934</width>
    <height>677</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <layout class="QHBoxLayout" name="horizontalLayout" stretch="4,1">
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_5" stretch="3,1">
        <item>
         <widget class="QTabWidget" name="tabWidget">
          <property name="currentIndex">
           <number>1</number>
          </property>
          <widget class="QWidget" name="u_i_tab">
           <attribute name="title">
            <string>I,U-curve</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_3">
            <item row="0" column="1">
             <widget class="QTabWidget" name="tabWidget_2">
              <property name="currentIndex">
               <number>0</number>
              </property>
              <widget class="QWidget" name="tab">
               <attribute name="title">
                <string>Graphical</string>
               </attribute>
               <layout class="QGridLayout" name="gridLayout_5">
                <item row="0" column="0">
                 <widget class="PlotWidget" name="u_i_pw"/>
                </item>
               </layout>
              </widget>
              <widget class="QWidget" name="tab_2">
               <attribute name="title">
                <string>Fit statistics</string>
               </attribute>
               <layout class="QGridLayout" name="gridLayout">
                <item row="0" column="0">
                 <widget class="QPlainTextEdit" name="fit_stats_u_i_tb"/>
                </item>
               </layout>
              </widget>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="mosfet_tab">
           <attribute name="title">
            <string>Mosfet</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_7">
            <item row="0" column="0">
             <widget class="PlotWidget" name="u_u_pw"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="p_r_tab">
           <attribute name="title">
            <string>P,R-curve</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_4">
            <item row="0" column="1">
             <widget class="PlotWidget" name="p_r_pw"/>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="tab_3">
           <attribute name="title">
            <string>Scan information</string>
           </attribute>
           <layout class="QGridLayout" name="gridLayout_6">
            <item row="0" column="0">
             <widget class="QPlainTextEdit" name="scan_info_tb"/>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="PlotWidget" name="max_pow_pw"/>
        </item>
       </layout>
      </item>
      <item>
       <layout class="QVBoxLayout" name="verticalLayout_2">
        <item>
         <widget class="QPushButton" name="scan_btn">
          <property name="text">
           <string>▶︎  start scan</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="mosfet_trim_cb">
          <property name="text">
           <string>Trim data to mosfet dynamic range prior to fitting</string>
          </property>
          <property name="checked">
           <bool>true</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QCheckBox" name="max_p_tracking_cb">
          <property name="text">
           <string>Active maximum power point tracking</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_2" stretch="0,0,0,0">
          <item>
           <widget class="QPushButton" name="autorange_btn">
            <property name="text">
             <string>Auto range</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_3">
            <item>
             <widget class="QLabel" name="label">
              <property name="text">
               <string>Initial output voltage</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="u_start_ib">
              <property name="text">
               <string>0</string>
              </property>
              <property name="maxLength">
               <number>10</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <widget class="QLabel" name="label_2">
              <property name="text">
               <string>Final output voltage</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QLineEdit" name="u_end_ib">
              <property name="text">
               <string>3.3</string>
              </property>
              <property name="maxLength">
               <number>10</number>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="verticalLayout_4"/>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_7">
          <item>
           <widget class="QLabel" name="label_3">
            <property name="text">
             <string>Number of samples</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="num_samples_ib">
            <property name="text">
             <string>100</string>
            </property>
            <property name="maxLength">
             <number>10</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_8">
          <item>
           <widget class="QLabel" name="label_4">
            <property name="text">
             <string>Number of times to retake each sample</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="repeat_ib">
            <property name="text">
             <string>2</string>
            </property>
            <property name="maxLength">
             <number>10</number>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer_3">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <layout class="QVBoxLayout" name="verticalLayout_9">
          <item>
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Selected device</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QComboBox" name="devices_cb"/>
          </item>
         </layout>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
        <item>
         <widget class="QPushButton" name="save_btn">
          <property name="text">
           <string>Save to CSV</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>934</width>
     <height>24</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>PlotWidget</class>
   <extends>QGraphicsView</extends>
   <header>pyqtgraph</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
